File: src/main/java/ma/fstg/gestionprojets/services/ReportingService.java
package ma.fstg.gestionprojets.services;
import ma.fstg.gestionprojets.dto.response.*;
import java.util.List;
public interface ReportingService {
    DashboardResponse getDashboard();
    List<PhaseResponse> getPhasesTermineesNonFacturees();
    List<PhaseResponse> getPhasesFactureesNonPayees();
    List<PhaseResponse> getPhasesPayees();
    List<ProjetResponse> getProjetsEnCours();
    List<ProjetResponse> getProjetsClotures();
}

File: gestion-projets-frontend/src/services/reportingService.ts
import api from '../utils/axiosConfig';

export const reportingService = {
  getDashboard: () => api.get('/reporting/tableau-de-bord'),
  getPhasesTermineesNonFacturees: () => api.get('/reporting/phases/terminees-non-facturees'),
  getPhasesFactureesNonPayees: () => api.get('/reporting/phases/facturees-non-payees'),
  getPhasesPayees: () => api.get('/reporting/phases/payees'),
  getProjetsEnCours: () => api.get('/reporting/projets/en-cours'),
  getProjetsClotures: () => api.get('/reporting/projets/clotures'),
};
